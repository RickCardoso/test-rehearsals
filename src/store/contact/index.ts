export * from './types';
export * from './contact';
export * from './actions';
export * from './selectors';
