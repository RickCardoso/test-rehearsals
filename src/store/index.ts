import { configureStore } from '@reduxjs/toolkit';
import { createWrapper, HYDRATE } from 'next-redux-wrapper';
import { AnyAction, combineReducers, PreloadedState, Store } from 'redux';
import { contactReducer } from './contact';
import heapMiddleware from './middlewares/heap';

const combinedReducer = combineReducers({
  contact: contactReducer,
});

//TODO: Find right types
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const reducer = (state: any, action: AnyAction): any => {
  if (action.type === HYDRATE) {
    return {
      // Attention! This will overwrite client state! Real apps should use proper reconciliation.
      ...state, // use previous state
      ...action.payload, // apply delta from hydration
    };
  } else {
    return combinedReducer(state, action);
  }
};

export const initStore = (preloadedState?: PreloadedState<unknown>): Store => {
  return configureStore({
    reducer,
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(heapMiddleware),
    preloadedState,
  });
};

export const wrapper = createWrapper(initStore);
