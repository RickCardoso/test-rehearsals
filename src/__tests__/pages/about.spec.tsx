import '@testing-library/jest-dom/extend-expect';

import { CopanProvider } from '@loft/copan-components';
import { render } from '@testing-library/react';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import messages from '../../lang/pt';
import About from '../../pages/about';
import { initStore } from '../../store';

describe('pages/about', () => {
  test('should render About page correctly', async () => {
    const { asFragment } = render(
      <IntlProvider locale="pt" messages={messages as any}>
        <Provider store={initStore()}>
          <CopanProvider>
            <About />
          </CopanProvider>
        </Provider>
      </IntlProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
