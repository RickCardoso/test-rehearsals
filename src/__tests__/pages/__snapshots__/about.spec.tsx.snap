// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/about should render About page correctly 1`] = `
<DocumentFragment>
  <header
    class="CopanHeader-root-2 CopanHeader-fixedHeader-3"
  >
    <div
      class="MuiGrid-root CopanGrid-container-14 lgMaxWidth MuiGrid-container"
    >
      <div
        class="MuiGrid-root CopanGrid-item-15 verticalNoPadding CopanHeader-gridContainer-4 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="CopanHeader-leftContainer-5"
        >
          <a
            aria-label="Ir para página inicial da Loft"
            class="CopanHeader-logo-6"
            href="/"
          >
            <svg
              class="MuiSvgIcon-root CopanHeader-mobileLogo-7"
              color="#FF774A"
              focusable="false"
              role="img"
              style="font-size: 32px;"
              viewBox="0 0 140 119"
            >
              <g
                fill-rule="evenodd"
              >
                <polygon
                  points="139.879 44.876 131.05 44.876 131.05 28.52 118.437 28.52 118.437 98.567 131.05 98.567 131.05 57.384 139.879 57.384"
                />
                <path
                  d="M90.6315856,119.840822 L78.018973,119.840822 L78.018973,27.6728524 C78.018973,12.9617512 90.087982,0.991235202 104.922937,0.991235202 L104.922937,13.4996113 C97.0425766,13.4996113 90.6315856,19.8576189 90.6315856,27.6728524 L90.6315856,119.840822 Z"
                />
                <path
                  d="M71.6580365,44.8761224 L112.073027,44.8761224 L108.497351,57.3844985 L71.6580365,57.3844985 C68.8358742,68.510984 58.6626658,76.7724816 46.5798919,76.7724816 C32.3225946,76.7724816 20.724036,65.2697789 20.724036,51.1303105 C20.724036,36.9908421 32.3225946,25.4881394 46.5798919,25.4881394 C58.6626658,25.4881394 68.8358742,33.749637 71.6580365,44.8761224 Z M32.0753874,51.1303105 C32.0753874,59.0618718 38.5822342,65.514943 46.5798919,65.514943 C54.5775495,65.514943 61.0843964,59.0618718 61.0843964,51.1303105 C61.0843964,43.1987492 54.5775495,36.7456779 46.5798919,36.7456779 C38.5822342,36.7456779 32.0753874,43.1987492 32.0753874,51.1303105 Z"
                />
                <path
                  d="M34.9076757,97.7300156 C15.6595676,97.7300156 -0.000252252253,82.1996158 -0.000252252253,63.1093321 L-0.000252252253,28.5199196 L12.6123604,28.5199196 L12.6123604,63.1093321 C12.6123604,75.3012464 22.6141622,85.2216395 34.9076757,85.2216395 L34.9076757,97.7300156 Z"
                />
              </g>
              <title>
                Loft
              </title>
            </svg>
            <svg
              class="MuiSvgIcon-root CopanHeader-desktopLogo-8"
              color="#FF774A"
              focusable="false"
              role="img"
              style="font-size: 40px;"
              viewBox="0 0 140 119"
            >
              <g
                fill-rule="evenodd"
              >
                <polygon
                  points="139.879 44.876 131.05 44.876 131.05 28.52 118.437 28.52 118.437 98.567 131.05 98.567 131.05 57.384 139.879 57.384"
                />
                <path
                  d="M90.6315856,119.840822 L78.018973,119.840822 L78.018973,27.6728524 C78.018973,12.9617512 90.087982,0.991235202 104.922937,0.991235202 L104.922937,13.4996113 C97.0425766,13.4996113 90.6315856,19.8576189 90.6315856,27.6728524 L90.6315856,119.840822 Z"
                />
                <path
                  d="M71.6580365,44.8761224 L112.073027,44.8761224 L108.497351,57.3844985 L71.6580365,57.3844985 C68.8358742,68.510984 58.6626658,76.7724816 46.5798919,76.7724816 C32.3225946,76.7724816 20.724036,65.2697789 20.724036,51.1303105 C20.724036,36.9908421 32.3225946,25.4881394 46.5798919,25.4881394 C58.6626658,25.4881394 68.8358742,33.749637 71.6580365,44.8761224 Z M32.0753874,51.1303105 C32.0753874,59.0618718 38.5822342,65.514943 46.5798919,65.514943 C54.5775495,65.514943 61.0843964,59.0618718 61.0843964,51.1303105 C61.0843964,43.1987492 54.5775495,36.7456779 46.5798919,36.7456779 C38.5822342,36.7456779 32.0753874,43.1987492 32.0753874,51.1303105 Z"
                />
                <path
                  d="M34.9076757,97.7300156 C15.6595676,97.7300156 -0.000252252253,82.1996158 -0.000252252253,63.1093321 L-0.000252252253,28.5199196 L12.6123604,28.5199196 L12.6123604,63.1093321 C12.6123604,75.3012464 22.6141622,85.2216395 34.9076757,85.2216395 L34.9076757,97.7300156 Z"
                />
              </g>
              <title>
                Loft
              </title>
            </svg>
          </a>
        </div>
        <div
          class="CopanHeader-navigationDesktop-13"
        >
          <nav
            class="CopanNavigation-navigation-128 CopanHeader-navigation-9 CopanHeader-navigationWithoutAccessory-10"
          >
            <ul
              class="CopanNavigation-baseList-129"
            >
              <li
                class="CopanNavigationItem-item-131"
              >
                <a
                  href="/"
                  tabindex="0"
                  target=""
                >
                  Home
                </a>
              </li>
              <li
                class="CopanNavigationItem-item-131"
              >
                <a
                  href="/contact"
                  tabindex="0"
                  target=""
                >
                  Contato
                </a>
              </li>
              <li
                class="CopanNavigationItem-item-131"
              >
                <a
                  href="/about"
                  tabindex="0"
                  target=""
                >
                  Sobre
                </a>
              </li>
              <li
                class="CopanNavigationItem-item-131 CopanNavigation-extraNavigation-130 hasSubItems more"
              >
                <span
                  class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-paragraph-md-145 MuiTypography-body1"
                  tabindex="0"
                >
                  Mais 
                  <svg
                    class="CopanIcon-black-183"
                    height="16px"
                    viewBox="0 0 24 24"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title>
                      CopanIcon
                    </title>
                    <path
                      d="M12 14.61L17.3124 9.28995C17.6985 8.90335 18.3244 8.90335 18.7104 9.28995C19.0965 9.67656 19.0965 10.3034 18.7104 10.69L12.699 16.71C12.3612 17.0483 11.8398 17.0906 11.4561 16.8369C11.4012 16.8007 11.3492 16.7584 11.301 16.71L5.28954 10.69C4.90349 10.3034 4.90349 9.67656 5.28954 9.28995C5.67559 8.90335 6.3015 8.90335 6.68755 9.28995L12 14.61Z"
                      fill="#191F23"
                    />
                  </svg>
                </span>
                <ul
                  class=""
                >
                  <li
                    class="CopanNavigationItem-item-131 more"
                  >
                    <a
                      href="/"
                      tabindex="0"
                      target=""
                    >
                      Home
                    </a>
                  </li>
                  <li
                    class="CopanNavigationItem-item-131 more"
                  >
                    <a
                      href="/contact"
                      tabindex="0"
                      target=""
                    >
                      Contato
                    </a>
                  </li>
                  <li
                    class="CopanNavigationItem-item-131 more"
                  >
                    <a
                      href="/about"
                      tabindex="0"
                      target=""
                    >
                      Sobre
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="CopanHeader-extraComponent-12"
        />
        <div
          class="CopanHeaderMobileControls-root-193"
        >
          <button
            class="CopanHeaderMobileControls-toggleMenuIcon-194 CopanHeaderMobileControls-hiddenUntilSm-196"
          >
            <svg
              class="CopanIcon-black-183"
              height="24px"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                CopanIcon
              </title>
              <path
                d="M5.5 5C4.67157 5 4 5.67157 4 6.5C4 7.32843 4.67157 8 5.5 8H18.5C19.3284 8 20 7.32843 20 6.5C20 5.67157 19.3284 5 18.5 5H5.5Z"
                fill="#191F23"
              />
              <path
                d="M4 11.5C4 10.6716 4.67157 10 5.5 10H18.5C19.3284 10 20 10.6716 20 11.5C20 12.3284 19.3284 13 18.5 13H5.5C4.67157 13 4 12.3284 4 11.5Z"
                fill="#191F23"
              />
              <path
                d="M4 16.5C4 15.6716 4.67157 15 5.5 15H18.5C19.3284 15 20 15.6716 20 16.5C20 17.3284 19.3284 18 18.5 18H5.5C4.67157 18 4 17.3284 4 16.5Z"
                fill="#191F23"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiGrid-root CopanGrid-container-14 lgMaxWidth makeStyles-withHeader-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root CopanGrid-item-15 MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-h2-139 MuiTypography-body1"
      >
        Sobre nós
      </h2>
      <picture
        class="CopanImage-root-197"
      >
        <source
          media="(max-width: 600px)"
          srcset=""
          type="image/webp"
        />
        <source
          media="(min-width: 601px)"
          srcset=""
          type="image/webp"
        />
        <source
          media="(max-width: 600px)"
          srcset=""
          type="image/jpeg"
        />
        <img
          alt="Equipe da Loft junta assistindo a uma apresentação da empresa"
          loading="lazy"
          src=""
          style="width: 100%;"
        />
      </picture>
      <div
        class="MuiPaper-root MuiExpansionPanel-root Mui-expanded MuiPaper-elevation1"
      >
        <div
          aria-controls="panel1d-content"
          aria-disabled="false"
          aria-expanded="true"
          class="MuiButtonBase-root MuiExpansionPanelSummary-root Mui-expanded"
          id="panel1d-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiExpansionPanelSummary-content Mui-expanded"
          >
            <p
              class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-paragraph-md-145 MuiTypography-body1"
            >
              Dúvida #1
            </p>
          </div>
        </div>
        <div
          class="MuiCollapse-container MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1d-header"
                id="panel1d-content"
                role="region"
              >
                <div
                  class="MuiExpansionPanelDetails-root"
                >
                  <p
                    class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-paragraph-md-145 MuiTypography-body1"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
                sit amet blandit leo lobortis eget. Lorem ipsum dolor sit amet, consectetur adipiscing
                elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiExpansionPanel-root MuiPaper-elevation1"
      >
        <div
          aria-controls="panel2d-content"
          aria-disabled="false"
          aria-expanded="false"
          class="MuiButtonBase-root MuiExpansionPanelSummary-root"
          id="panel2d-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiExpansionPanelSummary-content"
          >
            <p
              class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-paragraph-md-145 MuiTypography-body1"
            >
              Dúvida #2
            </p>
          </div>
        </div>
        <div
          class="MuiCollapse-container MuiCollapse-hidden"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel2d-header"
                id="panel2d-content"
                role="region"
              >
                <div
                  class="MuiExpansionPanelDetails-root"
                >
                  <p
                    class="MuiTypography-root CopanTypography-root-152 CopanTypography-primary-132 CopanTypography-paragraph-md-145 MuiTypography-body1"
                  >
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex,
                sit amet blandit leo lobortis eget. Lorem ipsum dolor sit amet, consectetur adipiscing
                elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="makeStyles-footer-245"
      >
        <a
          href="/about"
          title="About"
        >
          Sobre
        </a>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
